evennia/utils/tests/test_funcparser.py

```diff
@@ -216,12 +238,48 @@ def test_parse(self, string, expected):
         # print(f"time: {(t1-t0)*1000} ms")
         self.assertEqual(expected, ret)
 
-    def test_parse_raise(self):
+    @parameterized.expand(
+        (
+            "Test malformed This is $dummy(a, b) and $bar(",
+            "Test $funcNotFound()",
+        )
+    )
+    def test_parse_raise_unparseable(self, unparseable):
         """
         Make sure error is raised if told to do so.
 
         """
-        string = "Test malformed This is $dummy(a, b) and $bar("
+        with self.assertRaises(funcparser.ParsingError):
+            self.parser.parse(unparseable, raise_errors=True)
+
+    @patch("evennia.utils.funcparser._MAX_NESTING", 2)
+    def test_parse_max_nesting(self):
+        """
+        Make sure it is an error if the max nesting value is reached.
+
+        TODO: Does this make sense? When it sees the first function, len(callstack)
```

Comment:
Hm, that doesn't seem right, no. It should fail at 3 already. 
To be picky, this test does not show that you can nest 4 though, only that you can't nest 5 :)

Reply:
Fixed it, now it nests 4 and that's the minimum required to get the test to pass.